analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AttributeDataExtensionsTest.cs

```diff
@@ -70,5 +143,47 @@ private static AttributeData AttributeDataWithName(string attributeClassName)
             attributeData.Protected().Setup<INamedTypeSymbol>("CommonAttributeClass").Returns(namedType.Object);
             return attributeData.Object;
         }
+
+        private static AttributeData AttributeDataWithArguments(Dictionary<string, object> namedArguments = null, Dictionary<string, object> constructorArguments = null)
+        {
+            namedArguments ??= new();
+            constructorArguments ??= new();
+            var separator = (constructorArguments.Any() && namedArguments.Any()) ? ", " : string.Empty;
+            var code = $@"
+using System;
+
+public class MyAttribute: Attribute
+{{
+    public MyAttribute({constructorArguments.Select(x => $"{TypeName(x.Value)} {x.Key}").JoinStr(", ")})
+    {{
+    }}
+
+    {namedArguments.Select(x => $@"public {TypeName(x.Value)} {x.Key} {{ get; set; }}").JoinStr("\r\n")}
+}}
+
+[My({constructorArguments.Select(x => Quote(x.Value)).JoinStr(", ")}{separator}{namedArguments.Select(x => $"{x.Key}={Quote(x.Value)}").JoinStr(", ")})]
+public class Dummy {{ }}
+";
```

Comment:
This is much better, easier to understand :)
Nitpick
```suggestion
public class Dummy {{ }}";
```